backend/controller/SequenceController.ts

```diff
@@ -0,0 +1,378 @@
+import { ISequence } from '../../model/sequence/ISequence';
+import { ISlide } from '../../model/slide/ISlide';
+import { ISequenceWithSlides } from '../../model/sequence/ISequenceWithSlides';
+import { DBSequence } from '../../model/sequence/DBSequence';
+import { DBSlide } from '../../model/slide/DBSlide';
+import { LoernwerkError, LoernwerkErrorCodes } from '../loernwerkUtilities';
+import { DBUser } from '../../model/user/DBUser';
+/**
+ * Manages the sequence data in the database and handles inquiries requests regarding these
+ */
+export class SequenceController {
+    /**
+     * Creates a new sequence in the database with the given title and given user as owner.
+     * @param name the title of the sequence
+     * @param userId the userid of the owner
+     * @returns the created sequence
+     */
+    public static async createNewSequence(
+        name: string,
+        userId: number
+    ): Promise<ISequence> {
+        if (!this.isValidUser(userId)) {
+            throw new LoernwerkError(
+                'user not found',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        const dbSequence = new DBSequence();
+        let newCode = this.generateCode();
+        while ((await DBSequence.findBy({ code: newCode })).length > 0) {
+            newCode = this.generateCode();
+        }
+        dbSequence.code = newCode;
+        dbSequence.authorId = userId;
+        dbSequence.name = name;
+        dbSequence.slideCount = 0;
+        dbSequence.writeAccess = [];
+        dbSequence.readAccess = [];
+        await dbSequence.save();
+        return dbSequence;
+    }
+
+    /**
+     * Searches the database for a sequence with corresponding access code. Throws an error if no sequence could be found.
+     * @param code the code of the sequence
+     * @returns the sequence
+     */
+    public static async getSequenceByCode(code: string): Promise<ISequence> {
+        const dbSequence = await DBSequence.findOneBy({ code: code });
```

Comment:
for consistency, I'd also `upper` the code here, since this is another GET function